evalml/data_checks/invalid_targets_data_check.py

```diff
@@ -104,4 +107,12 @@ def validate(self, X, y):
                                                              message_code=DataCheckMessageCode.TARGET_BINARY_INVALID_VALUES,
                                                              details={"target_values": unique_values}).to_dict())
 
+        any_neg = not (y > 0).all() if y.dtype in numeric_dtypes else None
+        if any_neg and (self.objective_name in ['Root Mean Squared Log Error', 'Mean Squared Log Error', 'Mean Absolute Percentage Error']):
+            details = {"Count of offending values": sum(val <= 0 for val in y.values.flatten())}
+            messages["errors"].append(DataCheckError(message=f"Target has non-positive values which is not supported for {self.objective_name}",
+                                                     data_check_name=self.name,
+                                                     message_code=DataCheckMessageCode.TARGET_INCOMPATIBLE_OBJECTIVE,
+                                                     details=details).to_dict())
```

Comment:
@ParthivNaresh will this close #785? Looks like it will :)

Reply:
@dsherry I absolutely do believe it will!

Reply:
Very cool! You can update the PR description to say "Fix #1521 fix #785" and merging will close both :)